### PR TITLE
[JENKINS-54313] protect against null pointer where secrets can have no annotations

### PR DESCRIPTION
**Pull request by rawlingsj**
[JENKINS-54313](https://issues.jenkins-ci.org/browse/JENKINS-54313)

### CODE REVIEW

**Comment by jtnord** [comment]
The [annotation is mandatory](https://jenkinsci.github.io/kubernetes-credentials-provider-plugin/examples/) as it is used for display (human readable).  whilst it shouldn't blow up it shouldn't process the credential either.

**Comment by rawlingsj** [comment]
Just ran into this again when reinstalling our Jenkins X infra, this is the exception that this PR fixes:
```
Picked up _JAVA_OPTIONS: -Xmx400m
  11.412 [id=26]	SEVERE	jenkins.InitReactorRunner$1#onTaskFailed: Failed KubernetesCredentialProvider.startWatchingForSecrets
java.lang.NullPointerException
	at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.SecretUtils.getKeyName(SecretUtils.java:186)
	at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.SecretUtils.getNonNullSecretData(SecretUtils.java:167)
	at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.convertors.UsernamePasswordCredentialsConvertor.convert(UsernamePasswordCredentialsConvertor.java:49)
	at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.convertors.UsernamePasswordCredentialsConvertor.convert(UsernamePasswordCredentialsConvertor.java:37)
	at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.KubernetesCredentialProvider.convertSecret(KubernetesCredentialProvider.java:194)
	at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.KubernetesCredentialProvider.startWatchingForSecrets(KubernetesCredentialProvider.java:90)
Caused: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:104)
Caused: java.lang.Error
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:110)
	at hudson.init.TaskMethodFinder$TaskImpl.run(TaskMethodFinder.java:175)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:296)
	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1069)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:214)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Started
Running in Durability level: PERFORMANCE_OPTIMIZED
```

**Comment by grahamar** [comment]
I'm seeing this too, not all secrets are jenkins credentials, so annotations shouldn't be expected.

File: src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/SecretUtils.java [hunk 1]
```diff
@@ -200,11 +201,13 @@ public static String getNonNullSecretData(Secret s, String key, String exception
      * @return the custom mapping for the key or {@code key} (identical object) if there is no custom mapping.
      */
     public static String getKeyName(Secret s, String key) {
-        String customMapping = s.getMetadata().getAnnotations().get(JENKINS_IO_CREDENTIALS_KEYBINDING_ANNOTATION_PREFIX + key);
-        if (customMapping != null && customMapping.length() > 0) {
-            return customMapping;
+        Map<String, String> annotations = s.getMetadata().getAnnotations();
+        if (annotations != null && annotations.size() > 0){
+            String customMapping = s.getMetadata().getAnnotations().get(JENKINS_IO_CREDENTIALS_KEYBINDING_ANNOTATION_PREFIX + key);
```
Comment (jglick) [inline src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/SecretUtils.java]: Should use the `annotations` variable just introduced. Also there is no need to check the size:

```java
if (annotations != null) {
    String customMapping = annotations.get(JENKINS_IO_CREDENTIALS_KEYBINDING_ANNOTATION_PREFIX + key);
```

**Comment by jtnord** [comment]
[`BaseStandardCredentials`](https://github.com/jenkinsci/credentials-plugin/blob/11873056e05470405fa004adbd2967d96eeafa12/src/main/java/com/cloudbees/plugins/credentials/impl/BaseStandardCredentials.java#L86) cleans up nulls so that the description is not `null`   not sure what other credential types use but seems safe enough as we are calling the constructors for at least the types we support (and all of those extend from `BaseStandardCredentials` at some point.

**Pull request merged**